### PR TITLE
WIP: Keep First Gamepad Support

### PR DESCRIPTION
**Pull request by TheElixZammuto**
## Description
This PR adds an option to let Sunshine Perma-connect a gamepad on startup, reusing it if needed. This allows older games that have problems with controller hotplugging to work as expected across gaming sessions.
Tested and working on Windows, must be tested on Linux

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Dependency update (updates to dependencies)
- [ ] Documentation update (changes to documentation)
- [ ] Repository update (changes to repository files, e.g. `.github/...`)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

## Branch Updates
LizardByte requires that branches be up-to-date before merging. This means that after any PR is merged, this branch
must be updated before it can be merged. You must also
[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [x] I want maintainers to keep my branch updated

### CODE REVIEW

File: src/input.cpp [hunk 1]
```diff
@@ -686,6 +710,16 @@ class deinit_t : public platf::deinit_t {
 [[nodiscard]] std::unique_ptr<platf::deinit_t> init() {
   platf_input = platf::input();
 
+  if(config::input.keep_first_gamepad) {
+    //Create a dummy first gamepad that does not disconnect across sessions
```
Comment (cgutman) [inline src/input.cpp:714]: We should be connecting our dummy gamepads as part of the `launch` and `resume` processing according to the gamepad mask specified by the client in the `remoteControllersBitmap` parameter.

**Comment by FriendlyNeighborhoodShane** [comment]
Any updates on this?

**Comment by ConcurrentCrab** [comment]
@TheElixZammuto are you not working on this feature anymore? This doesn't seem to have been merged.

If so, would you mind if I try my hands at to carrying it based on your existing code?

**Comment by TheElixZammuto** [comment]
No, I did not like the approach and it will definitely have a merge conflict with some later commits (the PR is fairly old). Go ahead and make it yours!

**Comment by ConcurrentCrab** [comment]
Gotcha, thanks for the permission!

**Pull request closed without merging**